index.js

```diff
@@ -1,30 +1,194 @@
-main(input, 30);
+
+const generateFromQueue = async () => {
+    let data;
+    try {
+        data = await getMessage();
+    } catch (err) {
+        console.error('Unable to get message from queue:', err);
+        setTimeout(generateFromQueue, 10000);
```

Comment:
is this a supposed to be waiting for message forever? although i am not nodejs expert, it is not very readable. if you need to use error retry, maybe encapsulate just the getMessage function part?

Reply:
That makes a lot of sense to me. This part is intended to handle a situation where there is potentially an error accessing the queue itself so I think you are right to suggest that that should be encapsulated in getMessage. 

My plan for this worker would be to have it run in a loop, checking the queue, pulling messages etc which is why there are the recursive calls on success or errors but definitely let me know if there is another way to approach it, I'm very open to changing it.

Reply:
a snippet like below could give you some idea. it is cleaner, and still expressive enough to use for whatever purpose

```
const readFromQueue = async (queueUrl, successHandler, errorHandler) => {
  while (true) {
    try {
      const response = await sqs.receiveMessage({
        QueueUrl: queueUrl,
        MaxNumberOfMessages: 10, // Adjust as necessary
        WaitTimeSeconds: 20, // Long polling
      }).promise();

      if (response.Messages) {
        for (const message of response.Messages) {
          const parsedBody = JSON.parse(message.Body);
          if (parsedBody.type === 'success') {
            successHandler(parsedBody.data);
          } else if (parsedBody.type === 'error') {
            errorHandler(parsedBody.data);
          } else {
            console.warn(`Unknown message type: ${parsedBody.type}`);
          }

          // Delete the processed message from the queue
          await sqs.deleteMessage({
            QueueUrl: queueUrl,
            ReceiptHandle: message.ReceiptHandle
          }).promise();
        }
      }

    } catch (error) {
      console.error('Error reading from SQS:', error);
    }
  }
};

// Example usage:

const QUEUE_URL = 'YOUR_SQS_QUEUE_URL';

const successHandler = (data) => {
  console.log(`Success: ${data}`);
};

const errorHandler = (data) => {
  console.error(`Error: ${data}`);
};

readFromQueue(QUEUE_URL, successHandler, errorHandler);
```

Reply:
This is great! I really like the suggestion for a successHandler and an errorHandler. I've started implementing like those based off of your recommendataion [here](https://github.com/alex-808/flashcard-service/pull/1#discussion_r1295362532) and the code is much cleaner. I love that abstraction, thanks